examples/flow_tool/flow_tool.h

```diff
@@ -0,0 +1,209 @@
+                      HTTPResponseCode.InternalServerError);
+                } else {
+                  const db::Node* result = nullptr;
+                  if (path.empty()) {
+                    result = &Value(optional_root_node);
+                  } else {
+                    const db::Dir* ptr = &Value<db::Dir>(Value(optional_root_node).data);
+                    size_t i = 0;
+                    while (i < path.size() && ptr && !result) {
+                      const db::Dir* next_ptr = nullptr;
+                      for (const auto& e : ptr->dir) {
+                        const auto optional_next_node = fields.node[e];
+                        if (!Exists(optional_next_node)) {
+                          return Response(api::Error("InternalError",
+                                                     "The internal filesystem contains an invalid entry at index " +
+                                                         current::ToString(i) + " of path " + JSON(path) + "."),
+                                          HTTPResponseCode.InternalServerError);
+                        }
+                        const auto& next_node = Value(optional_next_node);
+                        if (next_node.name == path[i]) {
+                          if (i + 1u == path.size()) {
+                            result = &next_node;
+                            break;
+                          } else {
+                            if (!Exists<db::Dir>(next_node.data)) {
+                              return Response(
+                                  api::Error("BadRequest",
+                                             "Attempted to access a file as if it is a directory, at index " +
+                                                 current::ToString(i) + " of path " + JSON(path) + "."),
+                                  HTTPResponseCode.BadRequest);
+                            } else {
+                              next_ptr = &Value<db::Dir>(next_node.data);
+                              break;
+                            }
+                          }
+                        }
+                      }
+                      ptr = next_ptr;
+                      ++i;
+                    }
+                  }
+                  if (result) {
+                    struct ResponseGenerator {
+                      const ImmutableFields<storage_t>& fields;
+                      const std::vector<std::string>& path;
+                      ResponseGenerator(const ImmutableFields<storage_t>& fields, const std::vector<std::string>& path)
+                          : fields(fields), path(path) {}
+                      Response response;
+                      void FillProtoFields(api::FileOrDirResponse& proto_response_object) {
+                        proto_response_object.url = "smoke_test_passed://" + current::strings::Join(path, '/');
```

Comment:
smoke_test_passed? Should probably be a FlowTool constructor parameter.

Reply:
Why bother thinking of this if it will inevitably get fixed -- and unit-tested -- as development progresses?